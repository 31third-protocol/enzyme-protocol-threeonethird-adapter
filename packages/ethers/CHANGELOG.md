# @enzymefinance/ethers

## 4.0.10

### Patch Changes

- 1f61b9af0: Release 4.0.10

## 4.0.9

### Patch Changes

- f82ac29f9: Latest protocol version

## 4.0.8

### Patch Changes

- c06baeab1: Add eth_sign for Gnosis Safe

## 4.0.7

### Patch Changes

- 8bb3e9ff6: Publish packages

## 4.0.5

### Patch Changes

- 8fe3d4a57: Publishing latest release

## 4.0.4

### Patch Changes

- 84c2e211e: Lock bn.js dependency version
- c030c92cd: Aave claimRewardsToSelf selector

## 4.0.3

### Patch Changes

- e8d978204: Upgrading ethers.js

## 4.0.2

### Patch Changes

- b4fcf8bc: Update packages after audit feedback and fixes.

## 4.0.1

### Patch Changes

- 2b54b39f: Add devtools to monorepo
