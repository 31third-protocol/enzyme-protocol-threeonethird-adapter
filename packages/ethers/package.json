{
  "name": "@enzymefinance/ethers",
  "version": "4.0.4",
  "description": "Technology Regulated and Operated Investment Funds",
  "author": "Enzyme Council DAO <council@enzyme.finance>",
  "license": "MIT",
  "homepage": "https://enzyme.finance",
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/avantgardefinance/ethereum-devtools.git"
  },
  "bugs": {
    "url": "https://github.com/avantgardefinance/ethereum-devtools/issues"
  },
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "sideEffects": false,
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "require": "./dist/index.js"
    }
  },
  "scripts": {
    "build": "tsup",
    "clean": "rimraf .turbo .eslintcache node_modules dist",
    "lint": "eslint \"src/**/*.{js,ts}\" --cache",
    "format": "yarn lint --fix",
    "typecheck": "tsc --noEmit"
  },
  "dependencies": {
    "ethers": "^5.6.8"
  },
  "devDependencies": {
    "@types/jest": "^27.4.0",
    "jest": "^27.4.7",
    "tsup": "^5.11.13",
    "typescript": "^4.5.5"
  },
  "tsup": {
    "entry": [
      "src/index.ts"
    ],
    "format": [
      "esm",
      "cjs"
    ],
    "dts": true,
    "splitting": true,
    "clean": true
  },
  "eslintConfig": {
    "extends": "@enzymefinance/eslint-config",
    "rules": {
      "@typescript-eslint/no-shadow": "off"
    },
    "parserOptions": {
      "project": [
        "./tsconfig.json"
      ]
    }
  }
}
