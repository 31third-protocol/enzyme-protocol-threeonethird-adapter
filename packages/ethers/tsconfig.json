{
  "extends": "../../tsconfig.json",
  "include": ["src"],
  "files": [],
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@enzymefinance/hardhat/types": ["../hardhat/src/types.ts"]
    }
  }
}
