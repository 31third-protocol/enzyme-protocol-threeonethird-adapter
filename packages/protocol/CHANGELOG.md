# @enzymefinance/protocol

## 2.0.2

### Patch Changes

- 44bc0e32: adding curve and aave integration

## 2.0.1

### Patch Changes

- 8246e8ce: Initial release.
