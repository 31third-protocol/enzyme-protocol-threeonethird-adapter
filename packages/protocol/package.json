{
  "name": "@enzymefinance/protocol",
  "version": "4.0.7",
  "description": "Technology Regulated and Operated Investment Funds",
  "author": "Enzyme Council DAO <council@enzyme.finance>",
  "license": "GPL-3.0",
  "homepage": "https://enzyme.finance",
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "sideEffects": false,
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/index.mjs",
      "require": "./dist/index.js"
    }
  },
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/enzymefinance/protocol.git"
  },
  "bugs": {
    "url": "https://github.com/enzymefinance/protocol/issues"
  },
  "keywords": [
    "DeFi",
    "Ethereum",
    "Enzyme",
    "Enzyme Finance",
    "Smart Contracts",
    "Solidity"
  ],
  "scripts": {
    "build": "tsup",
    "test": "jest --maxWorkers 50%",
    "lint": "eslint \"{deploy,tasks,tests,src}/**/*.{js,ts}\" --cache",
    "format": "pnpm lint --fix",
    "typecheck": "tsc --noEmit"
  },
  "dependencies": {
    "@enzymefinance/ethers": "workspace:4.0.7",
    "@ethersproject/abi": "^5.7.0",
    "@ethersproject/abstract-signer": "^5.7.0",
    "decimal.js": "^10.3.1",
    "ethers": "^5.7.1"
  },
  "devDependencies": {
    "@enzymefinance/eslint-config": "1.0.26",
    "@jest/environment": "^28.1.3",
    "@nomiclabs/hardhat-ethers": "2.1.1",
    "@nomiclabs/hardhat-etherscan": "3.1.0",
    "@openzeppelin-solc-0.7/contracts": "npm:@openzeppelin/contracts@3.4.2-solc-0.7",
    "@openzeppelin/contracts": "3.4.1",
    "@types/fs-extra": "^9.0.13",
    "@types/jest": "^28.1.6",
    "@types/node": "^18.6.4",
    "@uniswap/v3-periphery": "github:uniswap/v3-periphery#80f26c86c57b8a5e4b913f42844d4c8bd274d058",
    "deepmerge": "^4.2.2",
    "dotenv": "^16.0.1",
    "eslint": "^8.12.0",
    "fs-extra": "^10.1.0",
    "glob": "^8.0.3",
    "hardhat": "2.11.2",
    "hardhat-contract-sizer": "^2.5.1",
    "hardhat-deploy": "^0.11.15",
    "jest": "^28.1.3",
    "jest-diff": "^28.1.3",
    "jest-environment-node": "^28.1.3",
    "jest-esbuild": "^0.2.9",
    "jest-matcher-utils": "^28.1.3",
    "jest-snapshot": "^28.1.3",
    "merkletreejs": "^0.2.31",
    "ts-node": "^10.7.0",
    "tsconfig-paths": "^4.1.0",
    "tsup": "^6.2.1",
    "typescript": "^4.5.5"
  },
  "tsup": {
    "entry": [
      "src/index.ts"
    ],
    "format": [
      "esm",
      "cjs"
    ],
    "dts": true,
    "splitting": true,
    "clean": true,
    "silent": true,
    "sourcemap": true
  },
  "eslintConfig": {
    "extends": "@enzymefinance/eslint-config",
    "rules": {
      "@typescript-eslint/no-shadow": "off"
    },
    "parserOptions": {
      "project": [
        "./tsconfig.json"
      ]
    }
  }
}
