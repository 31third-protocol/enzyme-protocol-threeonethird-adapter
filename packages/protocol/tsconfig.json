{
  "extends": "../../tsconfig.json",
  "include": ["src"],
  "files": [],
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@enzymefinance/ethers": ["../ethers/src"]
    }
  }
}
