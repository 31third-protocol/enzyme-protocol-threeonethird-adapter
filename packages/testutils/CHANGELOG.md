# @enzymefinance/testutils

## 3.0.2

### Patch Changes

- ccad45f0: Removed jest expect assertions from testutils package.

## 3.0.1

### Patch Changes

- e8654765: Publish solidity files.
- Updated dependencies [e8654765]
  - @enzymefinance/protocol@3.0.1

## 3.0.0

### Major Changes

- 65ce0f49: Publishing testutils alongside protocol package.

### Patch Changes

- Updated dependencies [65ce0f49]
  - @enzymefinance/protocol@3.0.0
