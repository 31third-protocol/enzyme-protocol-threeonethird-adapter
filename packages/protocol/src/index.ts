import { BigNumber, BigNumberish } from 'ethers';
import {
  contract,
  Call,
  Send,
  AddressLike,
  Contract,
} from '@crestproject/crestproject';

// Persistent core
export { Dispatcher } from './codegen/Dispatcher';
export { VaultProxy } from './codegen/VaultProxy';

// Persistent release interfaces
export { IMigrationHookHandler } from './codegen/IMigrationHookHandler';
export { IProxiableVault } from './codegen/IProxiableVault';

// Release core
export { FundDeployer } from './codegen/FundDeployer';
export { ComptrollerLib } from './codegen/ComptrollerLib';
export { ComptrollerProxy } from './codegen/ComptrollerProxy';
export { PermissionedVaultActionLib } from './codegen/PermissionedVaultActionLib';
export { VaultLib } from './codegen/VaultLib';
export { FundLifecycleLib } from './codegen/FundLifecycleLib';

// Infrastructure
export { Engine } from './codegen/Engine';
export { ValueInterpreter } from './codegen/ValueInterpreter';

// Extensions
export { IExtension } from './codegen/IExtension';
export { FeeManager } from './codegen/FeeManager';
export { IntegrationManager } from './codegen/IntegrationManager';
export { PolicyManager } from './codegen/PolicyManager';

// Primitive price feeds
export { ChainlinkPriceFeed } from './codegen/ChainlinkPriceFeed';

// Derivative price feeds
export { AggregatedDerivativePriceFeed } from './codegen/AggregatedDerivativePriceFeed';
export { ChaiPriceFeed } from './codegen/ChaiPriceFeed';
export { CompoundPriceFeed } from './codegen/CompoundPriceFeed';

// Integratee interfaces
export { IUniswapV2Factory } from './codegen/IUniswapV2Factory';
export { IUniswapV2Pair } from './codegen/IUniswapV2Pair';
export { IKyberNetworkProxy } from './codegen/IKyberNetworkProxy';
export { ICERC20 } from './codegen/ICERC20';

// Integration adapters
export { ChaiAdapter } from './codegen/ChaiAdapter';
export { EngineAdapter } from './codegen/EngineAdapter';
export { KyberAdapter } from './codegen/KyberAdapter';
export { TrackedAssetsAdapter } from './codegen/TrackedAssetsAdapter';
export { ZeroExV2Adapter } from './codegen/ZeroExV2Adapter';
export { CompoundAdapter } from './codegen/CompoundAdapter';

// Fees
export { IFee } from './codegen/IFee';
export { EntranceRateBurnFee } from './codegen/EntranceRateBurnFee';
export { EntranceRateDirectFee } from './codegen/EntranceRateDirectFee';
export { ManagementFee } from './codegen/ManagementFee';
export { PerformanceFee } from './codegen/PerformanceFee';

// Policies
export { IPolicy } from './codegen/IPolicy';
export { AdapterBlacklist } from './codegen/AdapterBlacklist';
export { AdapterWhitelist } from './codegen/AdapterWhitelist';
export { AssetBlacklist } from './codegen/AssetBlacklist';
export { AssetWhitelist } from './codegen/AssetWhitelist';
export { BuySharesPriceFeedTolerance } from './codegen/BuySharesPriceFeedTolerance';
export { MaxConcentration } from './codegen/MaxConcentration';
export { InvestorWhitelist } from './codegen/InvestorWhitelist';

// Peripheral
export { FundCalculator } from './codegen/FundCalculator';

// Mocks
export { MockVaultLib } from './codegen/MockVaultLib';
export { MockChaiIntegratee } from './codegen/MockChaiIntegratee';
export { MockChaiPriceSource } from './codegen/MockChaiPriceSource';
export { MockGenericAdapter } from './codegen/MockGenericAdapter';
export { MockGenericIntegratee } from './codegen/MockGenericIntegratee';
export { MockKyberIntegratee } from './codegen/MockKyberIntegratee';
export { MockKyberPriceSource } from './codegen/MockKyberPriceSource';
export { MockChainlinkPriceSource } from './codegen/MockChainlinkPriceSource';
export { MockToken } from './codegen/MockToken';
export { MockZeroExV2Integratee } from './codegen/MockZeroExV2Integratee';
export { WETH } from './codegen/WETH';

// prettier-ignore
export interface StandardToken extends Contract<StandardToken> {
  // Shortcuts (using function name of first overload)
  allowance: Call<(owner: AddressLike, spender: AddressLike) => BigNumber, Contract<any>>
  approve: Send<(spender: AddressLike, amount: BigNumberish) => boolean, Contract<any>>
  balanceOf: Call<(account: AddressLike) => BigNumber, Contract<any>>
  decimals: Call<() => BigNumber, Contract<any>>
  totalSupply: Call<() => BigNumber, Contract<any>>
  transfer: Send<(recipient: AddressLike, amount: BigNumberish) => boolean, Contract<any>>
  transferFrom: Send<(sender: AddressLike, recipient: AddressLike, amount: BigNumberish) => boolean, Contract<any>>

  // Explicit accessors (using full function signature)
  'allowance(address,address)': Call<(owner: AddressLike, spender: AddressLike) => BigNumber, Contract<any>>
  'approve(address,uint256)': Send<(spender: AddressLike, amount: BigNumberish) => boolean, Contract<any>>
  'balanceOf(address)': Call<(account: AddressLike) => BigNumber, Contract<any>>
  'decimals()': Call<() => BigNumber, Contract<any>>
  'totalSupply()': Call<() => BigNumber, Contract<any>>
  'transfer(address,uint256)': Send<(recipient: AddressLike, amount: BigNumberish) => boolean, Contract<any>>
  'transferFrom(address,address,uint256)': Send<(sender: AddressLike, recipient: AddressLike, amount: BigNumberish) => boolean, Contract<any>>
}

export const StandardToken = contract<StandardToken>()`
  event Approval(address indexed owner, address indexed spender, uint256 value)
  event Transfer(address indexed from, address indexed to, uint256 value)
  function allowance(address owner, address spender) view returns (uint256)
  function approve(address spender, uint256 amount) returns (bool)
  function balanceOf(address account) view returns (uint256)
  function decimals() view returns (uint8)
  function totalSupply() view returns (uint256)
  function transfer(address recipient, uint256 amount) returns (bool)
  function transferFrom(address sender, address recipient, uint256 amount) returns (bool)
`;
