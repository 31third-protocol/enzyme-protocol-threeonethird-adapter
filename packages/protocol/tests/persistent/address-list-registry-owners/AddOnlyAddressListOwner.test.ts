import type { AddressLike } from '@enzymefinance/ethers';
import { randomAddress } from '@enzymefinance/ethers';
import type { AddressListRegistry, ITestAddOnlyAddressListOwner } from '@enzymefinance/protocol';
import { AaveV2ATokenListOwner } from '@enzymefinance/protocol';
import type { ProtocolDeployment } from '@enzymefinance/testutils';
import { deployProtocolFixture } from '@enzymefinance/testutils';
import type { BigNumber } from 'ethers';

const invalidItem = randomAddress();
let fork: ProtocolDeployment;
let addressListRegistry: AddressListRegistry;
let listId: BigNumber, listOwner: ITestAddOnlyAddressListOwner;
let validItem: AddressLike;

// TODO: add Aave v3 when ready
const listKeys = ['aaveV2'];
describe.each(listKeys)('%s as list', (listKey) => {
  beforeEach(async () => {
    fork = await deployProtocolFixture();

    addressListRegistry = fork.deployment.addressListRegistry;

    listId = await addressListRegistry.getListCount();

    // list-specific vars
    // Deploy lists fresh so that no items are registered
    switch (listKey) {
      case 'aaveV2':
        validItem = fork.config.aaveV2.atokens.adai;
        listOwner = await AaveV2ATokenListOwner.deploy(
          fork.deployer,
          addressListRegistry,
          'Aave v2: aTokens',
          fork.config.aaveV2.lendingPoolAddressProvider,
        );
        break;
    }
  });

  describe('addValidatedItemsToList', () => {
    // TODO: really, we should use a mock of the valid token's interface so that this reverts closer
    // to the case that we actually want to test, i.e., the actual call to aave in the case of aTokens
    it('does not allow an invalid item', async () => {
      await expect(listOwner.addValidatedItemsToList([invalidItem])).rejects.toBeReverted();
    });

    it('happy path', async () => {
      // Item should initially be unregistered
      expect(await addressListRegistry.isInList(listId, validItem)).toBe(false);

      const receipt = await listOwner.addValidatedItemsToList([validItem]);

      // Item should be registered
      expect(await addressListRegistry.isInList(listId, validItem)).toBe(true);

      expect(receipt).toMatchGasSnapshot(listKey);
    });
  });
});
