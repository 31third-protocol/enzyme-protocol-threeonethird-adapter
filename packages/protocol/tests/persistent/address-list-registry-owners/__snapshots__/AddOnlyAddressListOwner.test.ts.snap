// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aaveV2 as list addValidatedItemsToList happy path: aaveV2 1`] = `100902`;
